container.go

```diff
@@ -20,6 +21,8 @@ const (
 )
 
 type container struct {
+	sync.RWMutex
```

Comment:
I'm not too excited about the locks being anonymous... Seems like the lock should have a name.